server/services/group.js

```diff
@@ -51,8 +51,25 @@ function addUser(group, userId) {
 function removeUser(group, userId) {
   const userIdSchema = Joi.string().hex().length(24);
 
+  let tmpGroup;
+
   return JoiHelper.validate(userId, userIdSchema)
-    .then(validatedUserObject => group.removeUser(validatedUserObject));
+    .then((validatedUserObject) => {
+      return group.removeUser(validatedUserObject);
+    })
+    .then((updatedGroup) => {
+      tmpGroup = updatedGroup;
+      return userService.removeGroupByUserId(userId, updatedGroup.name);
+    })
+    .then(() => tmpGroup)
+    .catch((err) => {
+      if(err.status== "404") {
```

Comment:
The logic for checking the status should go as deep as you can make it. In this case, that's in the model.